pkg/apis/resource/validation/validation.go

```diff
@@ -693,3 +711,51 @@ func validateMap[K ~string, T any](m map[K]T, maxSize int, validateKey func(K, *
 	}
 	return allErrs
 }
+
+func validateDeviceStatus(device resource.AllocatedDeviceStatus, fldPath *field.Path, allocatedDevices sets.Set[structured.DeviceID]) field.ErrorList {
+	var allErrs field.ErrorList
+	deviceID := structured.DeviceID{Driver: device.Driver, Pool: device.Pool, Device: device.Device}
+	if !allocatedDevices.Has(deviceID) {
+		allErrs = append(allErrs, field.Invalid(fldPath, deviceID, "must be an allocated device in the claim"))
+	}
+	allErrs = append(allErrs, validateSlice(device.Conditions, -1,
+		func(condition metav1.Condition, fldPath *field.Path) field.ErrorList {
+			var allErrs field.ErrorList
+			return allErrs
+		}, fldPath.Child("conditions"))...)
```

Comment:
This not really validating anything :) 

```suggestion
	allErrs = append(allErrs, apimachineryvalidation.ValidateConditions(device.Conditions, fldPath.Child("conditions"))...)
```